tests/Run-Tests.ps1

```diff
@@ -0,0 +1,269 @@
+<#
+.SYNOPSIS
+    Runs Functional Tests for Islandwood project.
+
+.PARAMETER TestDirectory
+    The path to test binaries
+
+.PARAMETER Platform
+    platform to run tests on. Valid values are Win32 or ARM
+
+.PARAMETER Config
+    Test configuration to use for the test. Valid values are Debug or Release
+
+.PARAMETER Device
+    IP or MAC address of the device to run tests on
+
+.PARAMETER ModuleFilter
+    Regex of Test Modules to look for
+
+.PARAMETER TestFilter
+    Test filter to use for the test. Supports wildcard characters
+
+.PARAMETER WTLOutputDirectory
+    Path to test output. If not set, no result file will be saved.
+
+.PARAMETER NoCopy
+    Switch to disable copying test to the device
+
+.PARAMETER RedirectTAEFErrors
+    Redirect TAEF errors/failures to stderr.  This is used to integrate with VSTS powershell task.
+
+.PARAMETER WTTLogPath
+    Path to find WTTLog dll
+
+.EXAMPLE
+    Desktop: Run-Tests.ps1 -TestType FunctionalTest
+             Run-Tests.ps1 -TestType UnitTest
+    Phone:  Run-Tests.ps1 -TestType FunctionalTest -Platform ARM
+            Run-Tests.ps1 -TestType UnitTest -Platform ARM
+
+#>
+param(
+    [Parameter(Mandatory=$true)]
+    [string][ValidateSet("FunctionalTest", "UnitTest")]
+    $TestType,
+
+    [string]$TestDirectory,
+
+    [string][ValidateSet("Win32", "ARM")]
+    $Platform = "Win32",
+
+    [string][ValidateSet("Debug", "Release")]
+    $Config = "Debug",
+
+    [string]$Device = "127.0.0.1",
+
+    [string]$ModuleFilter,
+    
+    [string]$TestFilter,
+
+    [string]$WTLOutputDirectory,
+
+    [switch]$NoCopy,
+
+    [switch]$RedirectTAEFErrors,
+
+    [string]$WTTLogPath
+)
+$script:exitCode = 0
+
+$TargetingDevice = ($Platform -eq "ARM")
+
+$outputFileName;
+if ($TestType -eq "FunctionalTest")
+{
+    $outputFileName = "FunctionalTestsResult.wtl"
+    if (!$ModuleFilter) 
+    {
+        $ModuleFilter = "FunctionalTests.dll"
+    }
+}
+else
+{
+    $outputFileName = "UnitTestsResult.wtl"
+    if (!$ModuleFilter) 
+    {
+        $ModuleFilter = "*.UnitTests.dll"
+    }
+}
```

Comment:
Sure.  I was originally going to get rid of run-functionatest.ps1 and run-unittest.ps1 but decided to keep them for the build defeinitions.


---
In reply to: [100863948](https://github.com/Microsoft/WinObjC/pull/1951#discussion_r100863948) [](ancestors = 100863948)